### PR TITLE
Regression test

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/8058

### CODE REVIEW

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
It'd be nice to test both pre-PHP 8.2 (function does not exist) and post-PHP 8.2 (function exists) behaviour.

File: tests/PHPStan/Rules/Methods/CallMethodsRuleTest.php [hunk 1]
```diff
@@ -2628,4 +2628,33 @@ public function testBug7600(): void
...
+		if (PHP_VERSION_ID > 80200) {
+			$this->markTestSkipped('Test requires PHP before 8.2');
+		}
+		$this->checkThisOnly = false;
+		$this->checkNullables = false;
+		$this->checkUnionTypes = false;
+		$this->checkExplicitMixed = false;
+		$this->analyse([__DIR__ . '/data/bug-8058.php'], [
+			[
+				'Call to an undefined method mysqli::execute_query().',
```
Comment (staabm) [inline tests/PHPStan/Rules/Methods/CallMethodsRuleTest.php:2654]: pretty interessting that only the method is undefined, but not the `\mysqli_execute_query` function

Comment (ondrejmirtes) [reply tests/PHPStan/Rules/Methods/CallMethodsRuleTest.php:2654]: There's a separate rule for that - CallToNonExistentFunctionRule.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**